dtfs-central-api/src/v1/controllers/utilisation-report-service/fee-record-correction/get-fee-record-correction-transient-form-data.controller/index.test.ts

```diff
@@ -0,0 +1,182 @@
+import httpMocks, { MockResponse } from 'node-mocks-http';
+import { ObjectId } from 'mongodb';
+import {
+  FeeRecordCorrectionTransientFormDataEntityMockBuilder,
+  RECORD_CORRECTION_REASON,
+  RecordCorrectionTransientFormData,
+  TestApiError,
+} from '@ukef/dtfs2-common';
+import { HttpStatusCode } from 'axios';
+import { Response } from 'express';
+import { GetFeeRecordCorrectionTransientFormDataRequest, GetFeeRecordCorrectionTransientFormDataResponse, getFeeRecordCorrectionTransientFormData } from '.';
+import { FeeRecordCorrectionTransientFormDataRepo } from '../../../../../repositories/fee-record-correction-transient-form-data-repo';
+import { FeeRecordRepo } from '../../../../../repositories/fee-record-repo';
+
+jest.mock('../../../../../repositories/fee-record-repo');
+jest.mock('../../../../../repositories/fee-record-correction-transient-form-data-repo');
+
+console.error = jest.fn();
+
+describe('get-fee-record-correction-transient-form-data.controller', () => {
+  describe('getFeeRecordCorrectionTransientFormData', () => {
+    const reportId = 1;
+    const feeRecordId = 2;
+
+    const tfmUserId = new ObjectId().toString();
+
+    const aValidRequestQuery = () => ({ reportId: reportId.toString(), feeRecordId: feeRecordId.toString(), userId: tfmUserId });
+
+    const mockFeeRecordExists = jest.fn();
+    const mockTransientFormDataFind = jest.fn();
+
+    let req: GetFeeRecordCorrectionTransientFormDataRequest;
+    let res: MockResponse<Response>;
+
+    beforeEach(() => {
+      FeeRecordRepo.existsByIdAndReportId = mockFeeRecordExists;
+      mockFeeRecordExists.mockReturnValue(true);
+
+      FeeRecordCorrectionTransientFormDataRepo.findByUserIdAndFeeRecordId = mockTransientFormDataFind;
+      mockTransientFormDataFind.mockResolvedValue(null);
```

Comment:
I'd prefer we don't have default mocks and instead mock in every test or in a describe with it's own before each that states the condition, since we are sometimes relying on these and sometimes not and it makes the test reading a little more confusing.

Reply:
Fair point, moved into the individual test cases where required thanks